pkg/storage/kubernetes.go

```diff
@@ -57,6 +58,31 @@ func NewKubernetesIPAM(containerID string, ipamConf whereaboutstypes.IPAMConfig)
 	if err != nil {
 		return nil, err
 	}
+	lockName := getNormalizedIpRangeStr(ipamConf.Range)
+	poolLock := &whereaboutsv1alpha1.IPPoolLock{
+		ObjectMeta: metav1.ObjectMeta{Name: lockName, Namespace: namespace},
+	}
+	// try creating ip pool lock, retry (if already exists) until it succeeds
+	// this makes serial access to ip pool across cluster wide and this might decrease
+	// the load considerably on k8 api server because of too many update calls, this can
+	// happen due to retries upon status conflict errors
+	for {
+		err = c.Create(ctx, poolLock)
+		if err != nil {
+			if errors.IsAlreadyExists(err) {
+				interval, _ := rand.Int(rand.Reader, big.NewInt(1000))
```

Comment:
range of sleep interval of 1000ms seems large for in cluster clients as per this link:
https://pkg.go.dev/k8s.io/client-go/util/retry?utm_source=godoc#RetryOnConflict
should this be changed to range of 100ms?